repaygithub/ui-tools
# CACTUS-269 :: Timed ETE Tests

Ticket: https://repayonline.atlassian.net/browse/CACTUS-269

Making tests for this was a huge pain. Both Jest and TestCafe have a global `test` variable, so at the time that tests are running, the `test` that is referenced when building the timed testcafe test is actually a reference to Jest's `test`. 🤦‍♂️ To get around this, I had to try to mock the `test` from TestCafe. Unfortunately, I didn't know how to do that effectively, which limits our testing capability here. For example, I can't `await` the test function being called in the mock, so I can't check anything that was logged to the console. If anyone has any ideas for how I could improve the mock, that'd be awesome. Who knows, maybe that'd help me figure out how to do this with function chaining.

EDIT: Credit & thank you to @mikesoltow for helping me figure out a better solution for the test mock!

I went ahead and set up a branch in cactus that uses this package and times a test in the `mock-ebpp` example, so that should make testing just a bit easier.


## Testing Instructions

1. Pull down this branch
2. Navigate to `modules/testing-tools`
3. Run `yarn link`
4. In the cactus repo, check out the following branch: `cactus-269-ete-test-timer`
5. In cactus, run `yarn install`
6. In cactus, navigate to `examples/mock-ebpp`
7. Run `yarn test` and observe the output of the timer.
